staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -639,11 +640,17 @@ func (s *store) List(ctx context.Context, key, resourceVersion string, pred stor
 		if err != nil {
 			return err
 		}
-		return s.versioner.UpdateList(listObj, uint64(returnedRV), next)
+		remainingItemCount := getResp.Count - pred.Limit
+		// getResp.Count counts in objects that do not match the pred.
+		// Instead of returning inaccurate count, return 0.
+		if !pred.Empty() {
+			remainingItemCount = 0
```

Comment:
I wonder if a different value (other than 0) should be returned which represents unknown remaining count.